include/LightGBM/dataset.h

```diff
@@ -138,6 +155,19 @@ class Metadata {
     weights_[idx] = value;
   }
 
+  /*!
+  * \brief Set Initial Scores for one record.  Note that init_score might have multiple columns and is stored in column format.
+  * \param idx Index of this record
+  * \param value Initial score values for this record, one per class
+  */
+  inline void SetInitScoreAt(data_size_t idx, const double* values) {
+    const auto nclasses = num_classes();
+    const double* val_ptr = values;
+    for (int i = idx; i < nclasses * num_data_; i += num_data_, ++val_ptr) {
```

Comment:
Does this function really set init_score for one record?

Reply:
Yes, there is an initial score per class, so there are multiple initial scores in the case of a multiclass run.  Seems better to have one function than one for 1 class, and another for multiclass.

Reply:
I am quite confused about nclass * num_data in this line. Do you mean to loop over all data?